msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java

```diff
@@ -586,7 +479,7 @@ public ImmutableValue unpackValue()
         MessageFormat mf = getNextFormat();
         switch (mf.getValueType()) {
             case NIL:
-                unpackNil();
+                readByte();
```

Comment:
Nil is 1-byte format and this is already checked with mf.getValueType(). readByte() is used for forwarding the buffer position.